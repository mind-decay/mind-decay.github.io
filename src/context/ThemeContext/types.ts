import { Nullable } from 'types';

export type ThemeType = 'light' | 'dark';

export type ThemeContextType = {
  theme: ThemeType;
  toggleTheme: () => void;
};

export const isThemeType = (theme: Nullable<string>): theme is ThemeType => {
  return typeof theme === 'string' && ['light', 'dark'].includes(theme);
};
