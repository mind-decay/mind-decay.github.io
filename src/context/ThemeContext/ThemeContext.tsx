import { ReactNode, createContext, useContext, useEffect, useState } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';

import { ThemeContextType, ThemeType, isThemeType } from './types';
import { Nullable } from 'types/helpers';

import { darkTheme, lightTheme } from 'styles/themes';

const ThemeContext = createContext<Nullable<ThemeContextType>>(null);

export const useTheme = (): ThemeContextType => {
  const context = useContext(ThemeContext);

  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }

  return context;
};

type ThemeProviderProps = {
  children: ReactNode;
};

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const [theme, setTheme] = useState<ThemeType>('dark');
  const isLightTheme = theme === 'light';

  const toggleTheme = () => {
    const updatedTheme = isLightTheme ? 'dark' : 'light';
    setTheme(updatedTheme);
    localStorage.setItem('theme', updatedTheme);
  };

  useEffect(() => {
    const savedTheme = localStorage.getItem('theme');
    const prefersLight = window.matchMedia && window.matchMedia('(prefers-color-scheme: light)').matches;
    const checkCurrentTheme = isThemeType(savedTheme);

    if (checkCurrentTheme) {
      setTheme(savedTheme);
    } else if (prefersLight) {
      setTheme('light');
    }
  }, []);

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      <StyledThemeProvider theme={isLightTheme ? lightTheme : darkTheme}>{children}</StyledThemeProvider>
    </ThemeContext.Provider>
  );
};
