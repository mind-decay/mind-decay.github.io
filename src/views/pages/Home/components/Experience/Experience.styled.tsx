import styled from 'styled-components';

import { screenBreakpoints } from 'styles';

export const ExperienceContainer = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  row-gap: 5.5rem;
  width: 100%;

  @media screen and (max-width: ${screenBreakpoints.tablet}px) {
    row-gap: 2rem;
  }
`;
