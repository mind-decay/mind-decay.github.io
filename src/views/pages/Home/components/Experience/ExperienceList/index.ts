export * from './ExperienceList';
export * from './ExperienceList.styled';
export * from './types';
export * from './Descriptions';
