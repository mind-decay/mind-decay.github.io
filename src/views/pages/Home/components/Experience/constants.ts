import {
  AgroClubDescription,
  ExperienceItem,
  FreelanceDescription,
  SdgDescription,
  YouDoDescription,
} from './ExperienceList';

import agroClubLogo from 'assets/images/AgroClubLogo.png';
import agroClubLogoWebp from 'assets/images/AgroClubLogo.webp';
import sdgLogo from 'assets/images/SdgLogo.png';
import sdgLogoWebp from 'assets/images/SdgLogo.webp';
import freelanceLogo from 'assets/images/FreelanceIcon.png';
import freelanceLogoWebp from 'assets/images/FreelanceIcon.webp';
import youDoLogo from 'assets/images/youDouLogo.png';
import youDoLogoWebp from 'assets/images/youDouLogo.webp';

export const experienceData: ExperienceItem[] = [
  {
    icons: [youDoLogo, youDoLogoWebp],
    title: 'Senior Frontend-разработчик в YouDo Бизнес',
    dateFrom: 'Апрель 2024',
    dateTo: 'По настоящее время',
    Description: YouDoDescription,
  },
  {
    icons: [agroClubLogoWebp, agroClubLogo],
    title: 'Senior Frontend-разработчик в Agro.Club',
    dateFrom: 'Октябрь 2022',
    dateTo: 'Февраль 2024',
    Description: AgroClubDescription,
  },
  {
    icons: [sdgLogoWebp, sdgLogo],
    title: 'Lead Frontend-разработчик в Sidewalk Developers Group',
    dateFrom: 'Май 2021',
    dateTo: 'Октябрь 2022',
    Description: SdgDescription,
  },
  {
    icons: [freelanceLogoWebp, freelanceLogo],
    title: 'Frontend-разработчик на фрилансе',
    dateFrom: 'Июнь 2019',
    dateTo: 'Май 2021',
    Description: FreelanceDescription,
  },
];
