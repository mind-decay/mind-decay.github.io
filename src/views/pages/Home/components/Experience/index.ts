export * from './Experience';
export * from './Experience.styled';
