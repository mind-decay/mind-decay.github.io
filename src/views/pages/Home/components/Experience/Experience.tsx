import { ExperienceContainer } from './Experience.styled';
import { Title } from 'views/components/Typography';
import {
  AgroClubDescription,
  ExperienceItem,
  ExperienceList,
  FreelanceDescription,
  SdgDescription,
} from './ExperienceList';

import agroClubLogo from 'assets/images/AgroClubLogo.png';
import agroClubLogoWebp from 'assets/images/AgroClubLogo.webp';
import sdgLogo from 'assets/images/SdgLogo.png';
import sdgLogoWebp from 'assets/images/SdgLogo.webp';
import freelanceLogo from 'assets/images/FreelanceIcon.png';
import freelanceLogoWebp from 'assets/images/FreelanceIcon.webp';

export const experienceData: ExperienceItem[] = [
  {
    icons: [agroClubLogoWebp, agroClubLogo],
    title: 'Senior Frontend-разработчик в Agro.Club',
    dateFrom: 'Октябрь 2022',
    dateTo: 'Февраль 2024',
    Description: AgroClubDescription,
  },
  {
    icons: [sdgLogoWebp, sdgLogo],
    title: 'Lead Frontend-разработчик в Sidewalk Developers Group',
    dateFrom: 'Май 2021',
    dateTo: 'Октябрь 2022',
    Description: SdgDescription,
  },
  {
    icons: [freelanceLogoWebp, freelanceLogo],
    title: 'Frontend-разработчик на фрилансе',
    dateFrom: 'Июнь 2019',
    dateTo: 'Май 2021',
    Description: FreelanceDescription,
  },
];

export const Experience = () => {
  return (
    <ExperienceContainer id="experience">
      <Title variant="h2" gradient="mainGradient">
        Опыт Работы
      </Title>

      <ExperienceList experienceData={experienceData} />
    </ExperienceContainer>
  );
};
