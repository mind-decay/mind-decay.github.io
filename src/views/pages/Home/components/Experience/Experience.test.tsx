import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { ThemeProvider } from 'context/ThemeContext';

import { Experience } from './Experience';

test('renders Experience section', () => {
  const { container } = render(
    <ThemeProvider>
      <Experience />
    </ThemeProvider>,
  );

  expect(container).toMatchSnapshot();
});
