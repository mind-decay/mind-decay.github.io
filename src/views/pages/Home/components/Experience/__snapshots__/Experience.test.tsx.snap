// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Experience section 1`] = `
<div>
  <section
    class="sc-beySPh cSdoRX"
    id="experience"
  >
    <h2
      class="sc-dmyCSP bKfxDg"
    >
      Опыт Работы
    </h2>
    <ul
      class="sc-gLLuof hrQZin"
    >
      <li
        class="sc-iBdnpw ezHqEe"
      >
        <div
          class="sc-fsYfdN hUaoJX"
        >
          <div
            class="sc-qZrbh gEguwx"
          >
            <div
              class="sc-jsEeTM cvuTQX"
            >
              <picture>
                <source
                  srcset="/src/assets/images/AgroClubLogo.webp"
                  type="image/webp"
                />
                <img
                  alt="Senior Frontend-разработчик в Agro.Club"
                  height="32"
                  loading="lazy"
                  src="/src/assets/images/AgroClubLogo.png"
                  width="32"
                />
              </picture>
            </div>
            <h3
              class="sc-hLQSwg loxgvM"
            >
              Senior Frontend-разработчик в Agro.Club
            </h3>
          </div>
          <p
            class="sc-jTQCzO gtaWj"
          >
            Октябрь 2022
             - 
            Февраль 2024
          </p>
        </div>
        <div
          class="sc-kFCroH iNDQYz"
        >
          <h4
            class="sc-eDLKkx kkxFzJ"
          >
            Обязанности:
          </h4>
          <ul
            class="sc-irLvIq faxKLD"
          >
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Участвовал в разработке 3-х проектов компании
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разрабатывал UI-kit используемый для проектов компании
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разрабатывал новый функционал проектов в связке с дизайнером и бэкэнд разработчиками
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Рефакторил и оптимизировал существующую кодовую базу
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Код ревью
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Писал тесты
            </li>
          </ul>
        </div>
        <div
          class="sc-kFCroH iNDQYz"
        >
          <h4
            class="sc-eDLKkx kkxFzJ"
          >
            Достижения:
          </h4>
          <ul
            class="sc-irLvIq faxKLD"
          >
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разработал новый проект в короткие сроки, после чего занимался его поддержкой и развитием
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Написал новые фронтовые нотификации - уменьшил количество кода, написал новые компоненты, расширил функционал, уменьшил бойлерплейт
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Обновил модальные окна и многократно упростил работу с ними
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Сделал более подробную систему сбора аналитики на фронте, работая в сотрудничестве с аналитиком внутри команды
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разработал логику и компоненты для работы с multi-step формами
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Привёл множество компонентов к новому дизайну и отрефакторил старый код
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разработал функционал интерактивного туториала, для простого онбординга новых пользователей
            </li>
          </ul>
        </div>
      </li>
      <li
        class="sc-iBdnpw ezHqEe"
      >
        <div
          class="sc-fsYfdN hUaoJX"
        >
          <div
            class="sc-qZrbh gEguwx"
          >
            <div
              class="sc-jsEeTM cvuTQX"
            >
              <picture>
                <source
                  srcset="/src/assets/images/SdgLogo.webp"
                  type="image/webp"
                />
                <img
                  alt="Lead Frontend-разработчик в Sidewalk Developers Group"
                  height="32"
                  loading="lazy"
                  src="/src/assets/images/SdgLogo.png"
                  width="32"
                />
              </picture>
            </div>
            <h3
              class="sc-hLQSwg loxgvM"
            >
              Lead Frontend-разработчик в Sidewalk Developers Group
            </h3>
          </div>
          <p
            class="sc-jTQCzO gtaWj"
          >
            Май 2021
             - 
            Октябрь 2022
          </p>
        </div>
        <div
          class="sc-kFCroH iNDQYz"
        >
          <h4
            class="sc-eDLKkx kkxFzJ"
          >
            Обязанности:
          </h4>
          <ul
            class="sc-irLvIq faxKLD"
          >
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Проектирование архитектуры проектов
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Координация работы членов команды frontend разработки
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разработка UI-kit для проектов компании
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Создание новых компонентов и страниц
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Коммуникация с руководством, с последующим описанием, декомпозицией задач и передачей их в исполнение другим участникам команды frontend разработки
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Код ревью
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Рефакторинг и оптимизация проектов клиентов компании
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Написание тестов для компонентов и страниц
            </li>
          </ul>
        </div>
        <div
          class="sc-kFCroH iNDQYz"
        >
          <h4
            class="sc-eDLKkx kkxFzJ"
          >
            Достижения:
          </h4>
          <ul
            class="sc-irLvIq faxKLD"
          >
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разработал простой и удобный UI-kit совместно с дизайнером внутри команды
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Написал с нуля 4 проекта
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Перевёл один из проектов заказчика с Vue2/Nuxt на Next.js
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Ввёл style guide для команды
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Лидил команду frontend разработки весь период работы
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Внедрил код ревью в процесс разработки проектов внутри компании
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Создал удобный workspace в Notion для хранения интересных и полезных ресурсов связанных с разработкой, для членов команды
            </li>
          </ul>
        </div>
      </li>
      <li
        class="sc-iBdnpw ezHqEe"
      >
        <div
          class="sc-fsYfdN hUaoJX"
        >
          <div
            class="sc-qZrbh gEguwx"
          >
            <div
              class="sc-jsEeTM cvuTQX"
            >
              <picture>
                <source
                  srcset="/src/assets/images/FreelanceIcon.webp"
                  type="image/webp"
                />
                <img
                  alt="Frontend-разработчик на фрилансе"
                  height="32"
                  loading="lazy"
                  src="/src/assets/images/FreelanceIcon.png"
                  width="32"
                />
              </picture>
            </div>
            <h3
              class="sc-hLQSwg loxgvM"
            >
              Frontend-разработчик на фрилансе
            </h3>
          </div>
          <p
            class="sc-jTQCzO gtaWj"
          >
            Июнь 2019
             - 
            Май 2021
          </p>
        </div>
        <div
          class="sc-kFCroH iNDQYz"
        >
          <h4
            class="sc-eDLKkx kkxFzJ"
          >
            Обязанности:
          </h4>
          <ul
            class="sc-irLvIq faxKLD"
          >
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Переговоры и уточнение требований с заказчиком
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Анализ макета для последующего создания библиотеки компонентов
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разрабатывал многостраничные сайты и лендинги на чистом HTML/CSS/JS, а также веб-приложения на React/TypeScript
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Занимался рефакторингом и оптимизацией уже существующих проектов
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Работал с несколькими иностранными студиями, в качестве внештатного работника
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Переводил старые сайты с jquery на современный HTML/CSS/JavaScript либо же React/Next.js
            </li>
          </ul>
        </div>
        <div
          class="sc-kFCroH iNDQYz"
        >
          <h4
            class="sc-eDLKkx kkxFzJ"
          >
            Достижения:
          </h4>
          <ul
            class="sc-irLvIq faxKLD"
          >
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разработал более 40 проектов на чистом HTML/CSS/JS
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Разработал более 60 проектов на React/TypeScript
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Успешно завершил более 100 заказов
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Активно работал с GSAP и другими библиотеками для создания комплексных анимаций
            </li>
            <li
              class="sc-csKJxZ jPnwFw"
            >
              Смог успешно закрепится на международным рынке фриланса, несмотря на конкуренцию
            </li>
          </ul>
        </div>
      </li>
    </ul>
  </section>
</div>
`;
